lib/puppet/provider/iis_virtual_directory/webadministration.rb

```diff
@@ -40,25 +41,32 @@ def create
   end
 
   def update
+    if (@resource[:ensure] == :absent)
```

Comment:
This still seems like it should be a separate ticket/fix as it doesn't haven anything to do with path validation, its fixing a core part of the provider that shouldn't be firing. I think we can let this stay in so long a new bug ticket is added for this to investigate why this provider is acting like you report.

Reply:
MODULES-6062 documents the update after destroy issue. In response to the request in an earlier version of this PR, I added `ErrorAction Stop` to these PowerShell commands. They will throw an error if called on a deleted virtual directory.